snuba/datasets/cdc/groupassignee.py

```diff
@@ -11,5 +15,7 @@ class GroupAssigneeDataset(Dataset):
     there is no issue in fixing the name.
     """
 
-    def __init__(self) -> None:
-        super().__init__(default_entity=EntityKey.GROUPASSIGNEE)
+    def get_all_entities(self) -> Sequence[Entity]:
```

Comment:
No need to redefined the `get_all_entities()` function inside every single dataset. I think you should pass the entity keys to the constructor of the base Dataset class, then add the `get_all_entities()` function there to return them.

Reply:
When you say pass entity keys to constructor of base, I'm interpreting that as defining a new attribute say `__all_entities` in our base Dataset, and initializing the keys in every dataset using `super().__init__()` which passes to the base constructor. Is that correct?

Reply:
Yes, basically exactly the same as how default entity is passed through today, but replace that with a list of all the entities instead of just one.

Reply:
Makes sense, this would be congruent with what we do with `PluggableDataset` (https://github.com/getsentry/snuba/blob/master/snuba/datasets/pluggable_dataset.py) and how we store Dataset related properties as attributes on the class itself .